### DISCUSSION TITLE
Parser for xit files in Rust (Nom)

### DISCUSSION THREAD

**Discussion by sum12** [jotaen/xit#39] [Show & tell]
Thanks for the well documented spec and loads of examples at /syntax-guide

https://github.com/sum12/nomxit 

This is (rust) library to parse xit files
The parser produces `vec` of items with checkboxes, tags, duedates and priority. The code is oragnized into modules to parse each of these with some test coverage.

I wanted to build a SPA with browser storage (using rust-wasm). However that needs a lot HTML/CSS command than I currently have. So will probably stop here. Maybe someone else can pick it up.

That was fun. Thank again.

**Comment by jotaen** [comment]
Looks like a fun exercise, thanks for sharing. Should you carry on with it at some point, please keep me posted!

**Comment by sum12** [comment]
Hey, I managed to get a local wepage with https://ace.c9.io/ and browser storage. The editor flags out divergence from syntax. So that part is done :tada: 
 
However I am not convinced that just editor with custom parser would suffice as good enough browser based todo SPA. I really like the design on the main page here xit.jotaen.net. Got any ideas about how it would/should look and behave like ?
